Person.java

```diff
@@ -0,0 +1,80 @@
+
+public class Person {
+    private String firstName;
+    private String lastName;
+    private int age;
+    private Person partner;
+
+    public Person(String firstName, String lastName, int age) {
+        this.firstName = firstName;
+        this.lastName = lastName;
+        this.age = age;
+        this.partner = null;
+    }
+
+    public String getFirstName() {
+        return firstName;
+    }
+
+    public void setFirstName(String firstName) {
+        this.firstName = firstName;
+    }
+
+    public String getLastName() {
+        return lastName;
+    }
+
+    public void setLastName(String lastName) {
+        this.lastName = lastName;
+    }
+
+    public int getAge() {
+        return age;
+    }
+
+    public void setAge(int age) {
+        this.age = age;
+    }
+
+    public Person getPartner() {
+        return partner;
+    }
+
+    public void setPartner(Person partner) {
+        this.partner = partner;
+    }
+
+    public boolean isRetired() {
+        if (this instanceof Man) {
+            return age >= 65;
+        } else if (this instanceof Woman) {
+            return age >= 60;
+        }
+        return false;
+    }
+
+    public void registerPartnership(Person partner) {
+        this.partner = partner;
+        if (this instanceof Man && partner instanceof Woman) {
```

Comment:
This should not be ther see recording of lesson 19 i gave example how it should work